src/widgets/pathsViewer.js

```diff
@@ -53,7 +53,9 @@ var FlatsealPathsViewer = GObject.registerClass(
 
     _remove(row) {
       this._box.remove(row);
-      row.destroy();
+      if (row.destroy) {
+        row.destroy();
```

Comment:
Note that `destroy` is not a thing in gtk4 (except with windows).

What are you trying to acomplish here? Remove a row from a listbox? In such a case there is gtk_list_box_remove (or gtk_box_remove).

Reply:
that's from the original code, it's basically resetting the widget to re-generate it from the data coming from the model (e.g. list of "other files").

Reply:
I just saw that you are already removiing the row correctly. There is no need for the destroy then, GTK gets rids of anything which goes out of scope.

Reply:
Yeah, I know, but in practice, it wasn't really doing that (it was leaking somehow), and it did require this extra step.

Reply:
Then someone is holding a reference to the row, probably a struct held it as one of its fields, idk. As I said previously `destroy` is not a thing anymore and any similar method is heavily discouraged.

Reply:
> Note that `destroy` is not a thing in gtk4 (except with windows).
> 
> What are you trying to acomplish here? Remove a row from a listbox? In such a case there is gtk_list_box_remove (or gtk_box_remove).

Okayy

Reply:
> Note that `destroy` is not a thing in gtk4 (except with windows).
> 
> What are you trying to acomplish here? Remove a row from a listbox? In such a case there is gtk_list_box_remove (or gtk_box_remove).

When I wrote that code, I didn't properly read the docs so I just... add the if sentence. ^^

Reply:
Sorry!!